server.h

```diff
@@ -58,19 +63,31 @@ class CTFOServer final {
  public:
   using Storage = CTFOStorage<SherlockStreamPersister>;
   using MidichloriansServer = current::midichlorians::server::MidichloriansHTTPServer<CTFOServer>;
+  using RESTStorage = RESTfulStorage<Storage,
+#ifdef USE_BASIC_HYPERMEDIA
+                                     current::storage::rest::Hypermedia>;
+#else
+                                     current::storage::rest::AdvancedHypermedia>;
+#endif
 
   explicit CTFOServer(const std::string& cards_file,
                       int port,
                       const std::string& storage_file,
                       int midichlorians_port,
                       const std::string& midichlorians_file,
+                      int rest_port,
+                      const std::string& rest_url_prefix,
                       const std::chrono::milliseconds tick_interval_ms,
                       const bool debug_print_to_stderr = false)
       : port_(port),
         midichlorians_server_(
             midichlorians_port ? midichlorians_port : port, *this, tick_interval_ms, "/ctfo/log", "OK\n"),
         debug_print_(debug_print_to_stderr),
-        storage_(storage_file) {
+        storage_(storage_file),
+        rest_(storage_,
+              rest_port ? rest_port : port,
+              "/ctfo/rest",
+              rest_url_prefix + ':' + current::ToString(rest_port) + "/ctfo/rest") {
```

Comment:
Can `rest_port` be zero if checked above: `rest_port ? rest_port : port`, so this might result in `"http://localhost:0/ctfo/rest"`? Or the check above is redundant.